### ISSUE TITLE
Included county contest info in DOS dashboard refresh

### ISSUE THREAD

**Issue by ranweiler** [FreeAndFair/ColoradoRLA#409]
See also: #379, #325.

Right now, in the DOS dashboard, we don't have a mapping from counties to contests in the DOS response. Similarly, the endpoint that describes the contests themselves does not include data about which county/counties in which the contest occurs. This means, using the existing DOS dashboard client requests, we are unable to adopt the approach suggested in #325.

I suggest one of two approaches to solving this problem:
1. For each county status in the `county_status` field of the DOS dashboard refresh response, include an array of the IDs of contests under audit. This approach is inefficient if we include multi-county contests, since in general, the client would then need to aggregate results across counties upon each refresh.
2. Include a new `contest_status` field in the DOS dashboard refresh response. The value of this field should be an object which maps contest IDs to a contest status object. A contest status object should include any information we want to display in the "Contest Updates" panel of the DOS dashboard, as described in #325.

My guess is that approach (1) would be easiest to implement in the short term. If we don't currently have any multi-county contests, then the inefficiency mentioned is less of a problem, though we still need to search through all known counties to find the one which has the contest we care about.

My preference is that we adopt approach (2), though it would require more server-side work. It is a more long-term solution, and makes the client-side implementation more obvious, efficient, and less error-prone.

**Comment by kiniry** [comment]
(2) sounds like the right way to go to me.

**Comment by dmzimmerman** [comment]
As it happens, the new audit engine that I hope to finish today basically solves #2 (tracking contest status on a per contest per county basis). It does not currently aggregate anything, though.

**Comment by dmzimmerman** [comment]
The problem is that we really have no such thing as a statewide "contest" right now. All contests are synthesized from county CVR uploads. Hence the lack of aggregation.

**Comment by ranweiler** [comment]
@dmzimmerman my preference is that we do aggregation server side, but I could make either work. It needs to happen somewhere. In the short term, we could just get the (2)-style mapping, aggregate client-side, and then push that to the server as a separate PR if it ends up being problematic.

**Comment by nealmcb** [comment]
A statewide "contest" is simply a contest whose name appears identically in each county.
Isn't aggregation server-side necessary to get the right calculations of margin etc for the rla? Or am I not understanding what you're aggregating and when?

**Comment by kiniry** [comment]
I have updated the PVS model to reflect (2).

**Comment by kiniry** [comment]
The problem is @nealmcb that we cannot guarantee that all counties will use precisely the same name (in a String.equals sense) for a contest that spans counties.  Recall all of the weird things we have seen in manifests and CVRs...

**Comment by nealmcb** [comment]
This issue has been noted repeatedly over the years to CDOS, and they assure us that since they are using SCORE to set up elections now, that contests will have the same text. Of course, one may not believe that, and want to allow for some dynamic hooks to fix things up, but we have had assurances in the past. Confirming with them after their state-wide mock election, which I hope was designed among other things to test this SCORE feature, would be good.

**Comment by dmzimmerman** [comment]
@ranweiler, as I think more about it, the (2)-style mapping already exists as endpoint `/contest/county?a&b&c&d` for contests for counties `a`, `b`, `c`, `d`. Is that somehow insufficient?

**Comment by dmzimmerman** [comment]
@nealmcb, that would in fact make aggregation easier. And setting up a query to get all the contests for a county is easy (as evidenced by the fact that it has existed for a while). 

If we know what we want to be in a "contest status" object, it's an easy thing to make one - `CountyContestResults` and `CountyContestComparisonAudit` are almost exactly those things, once the new audit engine is merged in.

**Comment by ranweiler** [comment]
Closed in #411.
